simtools/simtel/simtel_histograms.py

```diff
@@ -79,17 +75,17 @@ def list_of_histograms(self):
         list:
             List of histograms.
         """
-
-        self._list_of_histograms = []
-        for file in self._histogram_files:
-            with EventIOFile(file) as f:
-                for o in yield_toplevel_of_type(f, Histograms):
-                    try:
-                        hists = o.parse()
-                    except Exception:  # pylint: disable=broad-except
-                        self._logger.warning(f"Problematic file {file}")
-                        continue
-                    self._list_of_histograms.append(hists)
+        if self._list_of_histograms is None:
+            self._list_of_histograms = []
+            for file in self._histogram_files:
+                with EventIOFile(file) as f:
+                    for o in yield_toplevel_of_type(f, Histograms):
+                        try:
+                            hists = o.parse()
+                        except Exception:  # pylint: disable=broad-except
+                            self._logger.warning(f"Problematic file {file}")
+                            continue
```

Comment:
Related to previous comment about adding a test here.

Reply:
Do I see it correctly the EventIOFile raises on ValueErrors? If this is correct, you could be a bit more specific.

See here for the function definition: https://github.com/cta-observatory/pyeventio/blob/16a6061102512e6db5f1fa3f81886fb95c7f327e/src/eventio/base.py#L57

Reply:
Why is being tested above is the parsers (`o.parse`) inside `yield_toplevel_of_type`, which does not raise anything but a warning:

https://github.com/cta-observatory/pyeventio/blob/16a6061102512e6db5f1fa3f81886fb95c7f327e/src/eventio/search_utils.py#L20C1-L26C1

In principle we could even remove this try and except statement. I have not written these lines of code in the PR but I assume there was a reason for Raul including it, therefore I prefer to keep them.

Reply:
Also, I am not in favor of discussing here topics that are not strictly related to this PR. Let's keep it simple.

Reply:
Sorry, wrong link: is this histgram.parse? Does it raises any exceptions? I actually cannot see any in https://github.com/cta-observatory/pyeventio/blob/16a6061102512e6db5f1fa3f81886fb95c7f327e/src/eventio/histograms.py#L10

Just wondering - I think this is one of the few places in the code (only?) where we introduce a broad exception.

Reply:
I removed now the try and except